common/src/api/external/mod.rs

```diff
@@ -2355,6 +2355,10 @@ pub struct SwitchPortSettingsView {
     /// Link-layer discovery protocol (LLDP) settings.
     pub link_lldp: Vec<LldpLinkConfig>,
 
+    /// TX equalization settings.  These are optional, and most links will not
+    /// need them.
+    pub tx_eq: Vec<Option<TxEqConfig>>,
```

Comment:
Is this collection truly variable in length, or can we expect a fixed length of optional values?

Reply:
Until we add support for breakout connectors and multiple links per port, the length will always be 1.  I implemented this as a vector, as that is how all of the other per-link settings are represented in this struct.